pkg/manager/watcher.go

```diff
@@ -89,3 +92,120 @@ func (sm *Manager) servicesWatcher(ctx context.Context) error {
 	log.Warnln("Stopping watching services for type: LoadBalancer in all namespaces")
 	return nil
 }
+
+// This file handles the watching of node annotations for configuration, it will exit once the annotations are
+// present
+func (sm *Manager) annotationsWatcher() error {
+	// Use a restartable watcher, as this should help in the event of etcd or timeout issues
+	log.Infoln("Kube-Vip is waiting for annotations to be present on this node")
+	hostname, err := os.Hostname()
+	if err != nil {
+		return err
+	}
+
+	// TEST - REMOVE
+	hostname = "k8s.bgp01"
+
+	labelSelector := metav1.LabelSelector{MatchLabels: map[string]string{"kubernetes.io/hostname": hostname}}
+	listOptions := metav1.ListOptions{
+		LabelSelector: labels.Set(labelSelector.MatchLabels).String(),
+	}
+
+	rw, err := watchtools.NewRetryWatcher("1", &cache.ListWatch{
+		WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
+			return sm.clientSet.CoreV1().Nodes().Watch(context.Background(), listOptions)
+		},
+	})
+	if err != nil {
+		return fmt.Errorf("error creating annotations watcher: %s", err.Error())
+	}
```

Comment:
This is likely a good candidate to use a metadata only client/informer: https://godoc.org/k8s.io/client-go/metadata

Reply:
I'll have to investigate that 🧐